API/Services/Implementations/UserService.cs

```diff
@@ -29,4 +36,43 @@ public class UserService(IUserRepository repository) : IUserService
             OsuUsername = user.Player.Username
         };
     }
+
+    public async Task<IEnumerable<OAuthClientDTO>?> GetClientsAsync(int id) =>
+        _mapper.Map<IEnumerable<OAuthClientDTO>?>((await _userRepository.GetAsync(id))?.Clients?.ToList());
+
+    public async Task<IEnumerable<MatchSubmissionStatusDTO>?> GetSubmissionsAsync(int id) =>
+        _mapper.Map<IEnumerable<MatchSubmissionStatusDTO>?>((await _userRepository.GetAsync(id))?.SubmittedMatches);
+
+    public async Task<bool> RejectSubmissionsAsync(int id, int? verifierId, int? verificationSource)
+    {
+        IEnumerable<Match>? submissions = (await _userRepository.GetAsync(id))?.SubmittedMatches?.ToList();
+        if (submissions is null)
+        {
+            // Return in the affirmative if the user has no submissions
+            return true;
+        }
+
+        foreach (Match match in submissions)
+        {
+            match.VerificationStatus = (int)MatchVerificationStatus.Rejected;
+            match.VerifierUserId = verifierId;
+            match.VerificationSource = verificationSource;
+        }
+
+        return await _matchesRepository.UpdateAsync(submissions, verifierId) == submissions.Count();
+    }
+
+    public async Task<UserDTO?> UpdateScopesAsync(int id, IEnumerable<string> scopes)
+    {
+        User? user = await _userRepository.GetAsync(id);
+        if (user is null)
+        {
+            return null;
+        }
+
+        user.Scopes = scopes.ToArray();
+        await _userRepository.UpdateAsync(user);
+
+        return _mapper.Map<UserDTO>(user);
+    }
```

Comment:
I think some extra logic here is needed. Such as forcing lower case on all scopes. Consider also restricting which scopes can be added to users (i.e. client and system should never be added to a user). Also consider checking the scopes themselves, i.e. `administrator` is not a scope we use, so it shouldn't be possible to add to a user.